pages/dkp/kaptain/2.0.0/configuration/auto-cleanup-settings/index.md

```diff
@@ -0,0 +1,213 @@
+```
+kubectl get workflows.argoproj.io
+
+NAME                                          STATUS      AGE
+data-passing-btwn-componefjdf8-1-3068851699   Running     17s
+dsl-control-structures-rugqkrh-1-2276733026   Succeeded   111s
+dsl-control-structures-rugqkrh-2-2259955407   Succeeded   51s
+end-to-end-mnist-pipeline-mnrr6
+```
+Each step of the pipeline is implemented using a `Pod.`  Pipeline pods are not deleted as long as the workflow that created them is present. Without cleanup, your namespace can become filled with completed pods:
+```
+kubectl get pods -l workflows.argoproj.io/workflow=dsl-control-structures-rugqkrh-1-2276733026
+
+NAME                                                     READY   STATUS      RESTARTS   AGE
+dsl-control-structures-rugqkrh-1-2276733026-2018045073   0/2     Completed   0          4m11s
+dsl-control-structures-rugqkrh-1-2276733026-2405487652   0/2     Completed   0          3m40s
+dsl-control-structures-rugqkrh-1-2276733026-3461867059   0/2     Completed   0          3m51s
+dsl-control-structures-rugqkrh-1-2276733026-4042755208   0/2     Completed   0          4m1s
+```
+
+### Using Python DSL for setting Pipeline TTL 
+Kubeflow Pipelines provide a Python Domain Specific Language (DSL) that allows you to specify a time-to-live (TTL) for the submitted Pipeline. Here is an excerpt from the [Pipeline tutorial][pipeline-tutorial]:
+```python
+@dsl.pipeline(
+    name="End-to-End MNIST Pipeline",
+    description="A sample pipeline to demonstrate multi-step model training, evaluation, export, and serving",
+)
+def mnist_pipeline(
+    input_bucket: str = "tutorial",
+    ...
+):
+    train_and_serve(
+        input_bucket=input_bucket,
+        ...
+    )
+    ...
+    # TTL for the workflow to persist after completion (1 hour)
+    dsl.get_pipeline_conf().set_ttl_seconds_after_finished(60 * 60)
+```
+
+This setting specifies the `ttlSecondsAfterFinished` property in the Argo Workflow definition specifying the amount of time the workflow will persist before it is cleaned up by the dedicated controller. 
+
+### Setting global TTL for completed Pipelines
+
+Kaptain has a global configuration property that allows you to set the default TTL for all created Pipelines; Workflow objects are deleted after the specified amount of time.  The default value for this property is 24 hours. However, this property only affects the Pipeline API Server; the Argo Workflow controller does not use this property.
+While Notebook users can set this property via the DSL, they cannot specify a longer interval than the global property setting.  The Pipeline component always uses the smaller of the two specified TTL values between the DSL config and the global property.  Because workflow objects can be useful in debugging, we recommend choosing a conservative value for the global property value.
```

Comment:
```suggestion
While Notebook users can set this property via the DSL, they cannot specify a longer interval than the global property setting. The Pipeline component always uses the smaller of the two specified TTL values between the DSL config and the global property. Because workflow objects can be useful in debugging, we recommend choosing a conservative value for the global property value.
```